processout-sdk/src/main/java/com/processout/processout_sdk/CustomerActionHandler.java

```diff
@@ -0,0 +1,168 @@
+                FingerprintWebView.getSettings().setJavaScriptEnabled(true); // enable javascript
+
+                // Defining fallback request in case the fingerprint times out or is unavailable
+                final MiscGatewayRequest fallbackGwayRequest = new MiscGatewayRequest("{\"threeDS2FingerprintTimeout\":true}");
+                fallbackGwayRequest.setURL(customerAction.getValue());
+                HashMap<String, String> fallbackHeaders = new HashMap<String, String>();
+                fallbackHeaders.put("Content-Type", "application/json");
+                fallbackGwayRequest.setHeaders(fallbackHeaders);
+
+                // Setup the timeout handler
+                final Handler timeOutHandler = new android.os.Handler();
+                // Configuring the action on timeout
+                final Runnable fingerprintTimeoutClearer = new Runnable() {
+                    @Override
+                    public void run() {
+                        // Destroying the webview
+                        destroyWebView(FingerprintWebView);
+                        callback.shouldContinue(fallbackGwayRequest.generateToken());
+                    }
+                };
+
+                // Catch webview URL redirects
+                FingerprintWebView.setWebViewClient(new WebViewClient() {
+                    @Override
+                    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
+                        // Check if the current Android version is supporte
+                        String token = null;
+
+                        // We cancel the timeout handler
+                        timeOutHandler.removeCallbacks(fingerprintTimeoutClearer);
+
+                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+                            token = request.getUrl().getQueryParameter("token");
+                        }
+                        if (token == null) {
+                            // Android version not supported for fingerprinting
+                            token = fallbackGwayRequest.generateToken();
+                        }
+                        callback.shouldContinue(token);
+                        return super.shouldOverrideUrlLoading(view, request);
+                    }
+                });
+
+                // Start the timeout
+                timeOutHandler.postDelayed(fingerprintTimeoutClearer, TimeUnit.SECONDS.toMillis(10));
+
+                // Load the fingerprint URL
+                FingerprintWebView.loadUrl(customerAction.getValue());
+
+                // Add the webview to content
+                if (with instanceof Activity) {
+                    // We perform the fingerprint by displaying the hidden webview
+                    rootLayout.addView(FingerprintWebView);
+                } else {
```

Comment:
Consider using early returns/breaks instead of else statements

Reply:
This code is actually from a commit but has been removed in the end as `with` is always the current activity which contains the payment information